pkg/report/sarif.go

```diff
@@ -207,11 +211,18 @@ func toSarifErrorLevel(severity string) string {
 	}
 }
 
-func toPathUri(input string) string {
+func ToPathUri(input string) string {
 	var matches = pathRegex.FindStringSubmatch(input)
 	if matches != nil {
 		input = matches[pathRegex.SubexpIndex("path")]
 	}
+	ref, err := containerName.ParseReference(input)
+	if err == nil {
+		input = strings.TrimSuffix(input, ref.Identifier())
+		input = strings.TrimSuffix(input, "@")
+		input = strings.TrimSuffix(input, ":")
```

Comment:
what do you think about using regex?
e.g. fmt.Sprintf('(\S+):?@?%s?', ref.Identifier())

Reply:
thanks! I tried to use regex. 

but it doesn't look better or cleaner for me.

Reply:
Could you try this method or something like that? `Identifies` contains a tag or digest, but we want to remove it.

```suggestion
		input = ref.Context().RepositoryStr()
```

Reply:
oh, sure. missed it. thanks

Reply:
this method is too smart.
it parses `docker.io/my-organization/my-app:2c6912aee7bde44b84d810aed106ca84f40e2e29`  to `my-organization/my-app`, but we want `docker.io/my-organization/my-app`. isn't it?

Reply:
i'm trying another methods from `ref.Context()`

Reply:
The default repository can be omitted. `my-organization/my-app` is fine.

Reply:
`ref.Context().Name()`:
* `docker.io/my-organization/my-app:2c6912aee7bde44b84d810aed106ca84f40e2e29` => `index.docker.io/my-organization/my-app`.
* `alpine:latest` => `index.docker.io/library/alpine` instead of `alpine`

`ref.Context().String()` returns `.Name()`.

Reply:
Either works for me.